### PR TITLE
Add minimal support for some more I/O syscalls in standalone mode

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: tests/other/standalone_syscalls/test.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  // ioctl and fcntl fail
+  int i = ioctl(1, 0);
+  assert(i < 0);
+  int f = fcntl(1, 0);
+  assert(f < 0);
+  // write to standard streams works.
+  write(1, "hello, world!", 5);
+  write(1, "\n", 1);
+}
+
```
Comment (sbc100) [inline tests/other/standalone_syscalls/test.cpp]: remove trailing newline

File: tests/other/standalone_syscalls/test.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <assert.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+extern "C" ssize_t write(int fd, const void *buf, size_t count);
```
Comment (sbc100) [inline tests/other/standalone_syscalls/test.cpp]: This is defined in unistd already.

File: tests/test_other.py [hunk 3]
```diff
@@ -10355,6 +10355,13 @@ def test_errno_type(self):
 ''')
     run_process([PYTHON, EMCC, 'errno_type.c'])
 
+  def test_standalone_syscalls(self):
+    run_process([PYTHON, EMCC, path_from_root('tests', 'other', 'standalone_syscalls', 'test.cpp'), '-o', 'test.wasm'])
+    with open(path_from_root('tests', 'other', 'standalone_syscalls', 'test.out')) as f:
+      expected = f.read()
+      for engine in WASM_ENGINES:
+        self.assertEqual(run_js('test.wasm', engine), expected)
```
Comment (sbc100) [inline tests/test_other.py]: Should we have something in `run_js` that does this automatically, perhaps activated by a decorator?

File: system/lib/libc/musl/src/unistd/write.c [hunk 4]
```diff
@@ -4,5 +4,19 @@
...
+#if __EMSCRIPTEN__
+	__wasi_ciovec_t iov = {
+		.buf = buf,
+		.buf_len = count
+	};
+	size_t num;
+	__wasi_errno_t error =
+			__wasi_fd_write(fd, &iov, 1, &num);
+	if (error != __WASI_ERRNO_SUCCESS) {
+		return __wasi_syscall_ret(error);
```
Comment (sbc100) [inline system/lib/libc/musl/src/unistd/write.c]: Strange that we had __wasi_fd_write already but we were not using it here..

File: tests/test_other.py (hunk 3 above)
Comment (kripken) [reply tests/test_other.py]: That does what? Compare the output to an expected value? Or run in multiple engines?

**Pull request merged**